File: src/main/java/com/fouan/events/ZombieMoved.java
package com.fouan.events;

import com.fouan.actors.ActorId;
import com.fouan.zones.Zone;
import lombok.Getter;

@Getter
public final class ZombieMoved
        extends Event<ZombieMoved>
        implements ZoneEvent {

    private final ActorId zombieId;
    private final Zone zone;

    public ZombieMoved(int turn, ActorId zombieId, Zone zone) {
        super(turn);
        this.zombieId = zombieId;
        this.zone = zone;
    }
}

File: src/main/java/com/fouan/events/ZombieMoved.kt
package com.fouan.events

import com.fouan.actors.ActorId
import com.fouan.zones.Position

class ZombieMoved(
    turn: Int,
    override val actorId: ActorId,
    val position: Position
) : Event<ZombieMoved>(turn),
    ActorEvent,
    ActionEvent,
    ZoneEvent
